go/ephemeral/lib.go

```diff
@@ -441,7 +439,35 @@ func (e *EKLib) BoxLatestTeamEK(ctx context.Context, teamID keybase1.TeamID, rec
 		return nil, err
 	}
 	boxes, _, err := boxTeamEKForUsers(ctx, e.G(), usersMetadata, teamEK)
-	return &boxes, nil
+	return boxes, err
+}
+
+func (e *EKLib) PrepareNewTeamEK(ctx context.Context, teamID keybase1.TeamID, signingKey libkb.NaclSigningKeyPair, recipients []keybase1.UID) (sig string, boxes *[]keybase1.TeamEkBoxMetadata, newMetadata keybase1.TeamEkMetadata, myBox *keybase1.TeamEkBoxed, err error) {
+
+	// If we need a new teamEK let's just create it when needed, the new
+	// members will be part of the team and will have access to it via the
+	// normal mechanisms.
+	if teamEKNeeded, err := e.NewTeamEKNeeded(ctx, teamID); err != nil {
+		return "", nil, newMetadata, nil, err
+	} else if teamEKNeeded {
+		return "", nil, newMetadata, nil, nil
+	}
```

Comment:
It feels confusing that `PrepareNewTeamEK` might not, in fact, prepare a new teamEK. Maybe it would be clearer to hoist this logic into `go/teams/transactions.go`?

Reply:
i like that the logic is contained in a single place though. the callers just have to check if it's `nil` or not and act accordingly..

Reply:
Maybe add "maybe" to the name?

Reply:
leaving as is